build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/SetCompatibleVersionsTask.java

```diff
@@ -45,11 +45,12 @@ public void executeTask() throws IOException {
         if (versionIds.isEmpty()) {
             throw new IllegalArgumentException("No version ids specified");
         }
+
         Integer transportVersion = versionIds.get(TRANSPORT_VERSION_TYPE);
         if (transportVersion == null) {
-            throw new IllegalArgumentException("TransportVersion id not specified");
+            // the release being done does not have transport versions - don't need to update CCS version
+            return;
```

Comment:
It seems this case could also occur if there's a configuration problem in Gradle. Is there perhaps another way we could check if the version _should_ be supplied rather than just whether it _was_ supplied?

Reply:
For example, if `TransportVersion.java` doesn't exist, that seems like a reliable indicator that the task doesn't have anything to do as opposed to a configuration bug.

Reply:
This is running on `main` (which will always have transport versions), updating with info from a release happening somewhere else. If that release is <8.8, there will not be a transport version to update.

More generally, this task should be run to update 'some versions' for the release that has just happened. This currently is just CCS version, but could be others in the future. The ccs version only needs to be updated if the release is for the current major version (eg for v9 releases, we don't update the version for any v8 releases, even though that also has a transport version). So maybe instead we need to pass in the major release number, so that the task can decide what (if anything) needs doing with the information it has. Then it can fail if its for a release of the same major version, but there's no transport version provided.

Reply:
See https://github.com/elastic/elasticsearch/pull/111489 for this alternative